### PR TITLE
Integrate frontend with new stoploss contract

### PR DESCRIPTION
**Pull request by yvesfracari**
Integrate order creation

- Valid to parameter with max validity of 6 months

Integrate new ponder api
- Remove calls to cow orderbook api

### CODE REVIEW

File: src/app/[chainId]/[safeAddress]/[orderId]/page.tsx [hunk 1]
```diff
@@ -12,6 +12,9 @@ export default function OrderPage({
     orderId: string;
   };
 }) {
+  if (params.chainId != 11155111) {
```
Comment (ribeirojose) [inline src/app/[chainId]/[safeAddress]/[orderId]/page.tsx:15]: 👀

File: src/components/ConsolidatedOrdersTable/index.tsx [hunk 2]
```diff
@@ -58,7 +60,11 @@ export function ConsolidatedOrdersTable() {
     columns,
     filterFields,
     enableRowSelection: (row) =>
-      row.original.status !== "filled" && row.original.status !== "cancelled",
+      [
+        OrderStatus.DRAFT,
+        OrderStatus.OPEN,
+        OrderStatus.PARTIALLY_FILLED,
+      ].includes(row.original.status),
```
Comment (ribeirojose) [inline src/components/ConsolidatedOrdersTable/index.tsx:67]: 👌

File: src/lib/tooltipDescriptions.ts [hunk 3]
```diff
@@ -14,7 +14,9 @@ export enum TOOLTIP_DESCRIPTIONS {
   ORDER = "The amount sold/bought. Also surplus for this order. This is the positive difference between the initial limit price and the actual (average) execution price.",
   ORACLE_TOKEN_SELL = "The oracle that will provide the price of the token being sold to the smart-contract. Both oracles must have the same quote currency.",
   ORACLE_TOKEN_BUY = "The oracle that will provide the price of the token being sold to the smart-contract. Both oracles must have the same quote currency.",
-  RELATED_ORDERS = "The orders created based on your initial request",
+  RELATED_ORDER = "The order posted on the CoW orderbook that is related to this stop loss order.",
```
Comment (ribeirojose) [inline src/lib/tooltipDescriptions.ts:17]:
```suggestion
  RELATED_ORDER = "The order posted on the CoW Orderbook that is related to this Stop Loss order.",
```

what does this mean though?

File: src/lib/tooltipDescriptions.ts [hunk 4]
```diff
@@ -14,7 +14,9 @@ export enum TOOLTIP_DESCRIPTIONS {
...
   MAX_TIME_SINCE_LAST_ORACLE_UPDATE = "The maximum time since the last oracle update. If the oracle has not been updated in this time, the order will not be executed.",
   TYPE = "The order can be fill or kill (entire order must be filled) or partial fill (part of the order can be filled respecting the limit price).",
+  VALID_TO = "The date and time at which the order will expire if not filled.",
+  VALID_TO_INPUT = "The time at which the order can be filled. After this time, the order will expire. This not take into account the transaction signing time.",
```
Comment (ribeirojose) [inline src/lib/tooltipDescriptions.ts:21]:
```suggestion
  VALID_TO_INPUT = "The absolute date and time before which the order can be filled. After this time, the order will expire.",
```

File: src/components/ReviewOrdersDialog.tsx [hunk 5]
```diff
@@ -195,6 +196,18 @@ function OrderContent({ order }: { order: DraftOrder }) {
         <OrderInformation title="Type" tooltipText={TOOLTIP_DESCRIPTIONS.TYPE}>
           {order.partiallyFillable ? "Partial fillable" : "Fill or Kill"}
         </OrderInformation>
+        <OrderInformation
+          title="Expiration date"
+          tooltipText={TOOLTIP_DESCRIPTIONS.VALID_TO}
+        >
+          {validToDate.toLocaleTimeString("en-GB", {
```
Comment (ribeirojose) [inline src/components/ReviewOrdersDialog.tsx:203]: why use `en-GB` here? Also I think we already have a function to format dates like this which we use in the table

File: src/components/OrderDetails.tsx [hunk 6]
```diff
@@ -302,7 +309,7 @@ export function OrderDetails({
               <InfoTooltip text={Number(limitPrice).toFixed(18)} /> {priceUnit}
             </div>
           </OrderDetailsInformation>
-          {(order?.filledPct || 0) > 0 && (
+          {(Number(order?.stopLossData?.filledPctBpt) || 0) > 0 && (
```
Comment (ribeirojose) [inline src/components/OrderDetails.tsx:312]: why bpt? Did you mean `bps` as in basis points?

File: src/lib/ponderApi/fetchOrders.ts [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+export const fetchOrderStatus = (
+  order: IStopLossPonder,
+  orderCanceled: boolean,
+): OrderStatus => {
+  if (!order?.stopLossData) return OrderStatus.OPEN;
+
+  const filledPctBpt = BigInt(order.stopLossData.filledPctBpt as bigint);
+  const isValid = Number(order.stopLossData.validTo) * 1000 > Date.now();
+
+  if (filledPctBpt >= BigInt(1000000)) return OrderStatus.FULFILLED;
```
Comment (ribeirojose) [inline src/lib/ponderApi/fetchOrders.ts:93]:
```suggestion
  if (filledPctBpt >= BigInt(1_000_000)) return OrderStatus.FULFILLED;
```
not sure I follow this though -- is 1_000_000 supposed to correspond to 100%?

File: src/lib/ponderApi/fetchOrders.ts [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+      address: COMPOSABLE_COW_ADDRESS,
+      abi: composableCowAbi,
+      functionName: "singleOrders",
+      args: [ownerAddress, orderHash],
+    })),
+  });
+  return multicallResults.map((result) => !result?.result);
+};
+
+export const fetchOrderStatus = (
```
Comment (ribeirojose) [inline src/lib/ponderApi/fetchOrders.ts:84]: you're not fetching anything here right?

File: src/app/[chainId]/[safeAddress]/[orderId]/page.tsx (hunk 1 above)
Comment (yvesfracari) [reply src/app/[chainId]/[safeAddress]/[orderId]/page.tsx:15]: For now I didn't deployed the new contract on other chains, so I am disable other chains

File: src/lib/ponderApi/fetchOrders.ts (hunk 7 above)
Comment (yvesfracari) [reply src/lib/ponderApi/fetchOrders.ts:93]: yes! That is right

File: src/components/OrderDetails.tsx (hunk 6 above)
Comment (yvesfracari) [reply src/components/OrderDetails.tsx:312]: yes, I will change this

File: src/lib/tooltipDescriptions.ts (hunk 4 above)
Comment (yvesfracari) [reply src/lib/tooltipDescriptions.ts:21]: Actually this input is a select with time periods (6 months, 1 month, etc). Think that your suggestion would be better for a exact time. What you say about

`How long the order will be valid and can be filled starting now. After this, the order will expire.`

File: src/components/ReviewOrdersDialog.tsx (hunk 5 above)
Comment (yvesfracari) [reply src/components/ReviewOrdersDialog.tsx:203]: I think that this was the function that we used there

File: src/lib/tooltipDescriptions.ts (hunk 3 above)
Comment (yvesfracari) [reply src/lib/tooltipDescriptions.ts]: Basically the order uid

**Pull request merged**